File: src/main/java/com/vaadin/demo/component/card/CardHeader.java
package com.vaadin.demo.component.card;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.card.Card;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.html.H2;
import com.vaadin.flow.router.Route;

@Route("card-header")
public class CardHeader extends Div {
    public CardHeader() {
        // tag::snippet[]
        Card card = new Card();

        Div header = new Div();
        header.getStyle().set("display", "flex")
                .set("flex-direction", "column-reverse")
                .set("line-height", "1.25");

        H2 title = new H2("Lapland");

        Div subtitle = new Div("The Exotic North");
        subtitle.getStyle().set("font-size", "0.8125rem").set("text-transform",
                "uppercase");

        header.add(title, subtitle);
        card.setHeader(header);

        card.add("Lapland is the northern-most region of Finland and an active outdoor destination.");
        // end::snippet[]

        card.setMaxWidth("300px");
        add(card);
    }

    public static class Exporter extends DemoExporter<CardHeader> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/card/card-header.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/card';
import { css, html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('card-header')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-card>
        <!-- tag::[] -->
        <div slot="header" style="display: flex; flex-direction: column-reverse; line-height: 1.25">
          <h2>Lapland</h2>
          <div style="font-size: 0.8125rem; text-transform: uppercase">The Exotic North</div>
        </div>
        <!-- end::[] -->
        <div>Lapland is the northern-most region of Finland and an active outdoor destination.</div>
      </vaadin-card>
      <!-- end::snippet[] -->
    `;
  }

  static styles = css`
    vaadin-card {
      max-width: 300px;
    }
  `;
}
